direct/io-bulk-fs/src/main/java/cz/o2/proxima/direct/bulk/Path.java

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright 2017-2020 O2 Czech Republic, a.s.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cz.o2.proxima.direct.bulk;
+
+import cz.o2.proxima.annotations.Internal;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.Serializable;
+import java.time.Duration;
+
+/** Proxima's abstraction of path in {@link FileSystem}. */
+@Internal
+public interface Path extends Serializable {
+
+  /**
+   * Create a Path representation of the given {@link File local file}.
+   *
+   * @param fs local filesystem
+   * @param path the local file
+   * @return Path representation of this local file
+   */
+  static Path local(FileSystem fs, File path) {
+    return new Path() {
+
+      @Override
+      public InputStream reader() throws IOException {
+        return new FileInputStream(path);
+      }
+
+      @Override
+      public OutputStream writer() throws IOException {
```

Comment:
What if file already exists (we want to make sure files are immutable accross all FS impls) / is directory?

Reply:
Then this should throw exception, do you see issues with that?